### PR TITLE
[EGD-3411] OneLine Text Specialization

### PR DESCRIPTION
**Pull request by PrzeBrudny**

### CODE REVIEW

File: module-gui/gui/widgets/text/core/cursors/TextInLineCursor.cpp [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            getDocument(), startDrawBlockPosition, startDrawBlockNumber, text->getTextFormat().getFont());
+        --moveCursor;
+
+        text->lines->drawStartConditions = {moveCursor.getBlockNumber(), moveCursor.getPosition()};
+        text->drawLines();
+    }
+
+    bool TextInLineCursor::checkInLineRightMove()
+    {
+        return onScreenPosition + 1 > text->lines->getLine()->length() &&
```
Comment (mjan-dev) [inline module-gui/gui/widgets/text/core/cursors/TextInLineCursor.cpp:38]: Minor. Could be without extra addition: 
```suggestion
        return onScreenPosition >= text->lines->getLine()->length() &&
```

File: module-gui/test/test-catch-text/test-gui-TextInLineCursor.cpp [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+
+    SECTION("Space for 10 sings - navigation with left ellipsis")
+    {}
+
+    SECTION("Space for 10 sings - navigation with right ellipsis")
+    {}
+
+    SECTION("Space for 10 sings - navigation with both ellipsis")
+    {}
+}
```
Comment (wojtekrzepecki) [inline module-gui/test/test-catch-text/test-gui-TextInLineCursor.cpp:147]: newline

**Review by wojtekrzepecki — APPROVED** [review]

File: module-gui/gui/widgets/text/core/lines/Lines.cpp [hunk 3]
```diff
@@ -38,13 +39,23 @@ namespace gui
...
-        drawMultiLine(drawCursor, w, h, lineYPosition, lineXPosition);
+        drawType == TextType::MultiLine ? drawMultiLine(drawCursor, w, h, lineYPosition, lineXPosition)
+                                        : drawSingleLine(drawCursor, w, h, lineYPosition, lineXPosition);
     }
 
     auto Lines::drawSingleLine(
         BlockCursor &drawCursor, Length w, Length h, Position lineYPosition, Position lineXPosition) -> void
     {
-        // TODO in EGD-3411
+        Length initHeight = text->getTextFormat().getFont()->info.line_height;
```
Comment (mjan-dev) [inline module-gui/gui/widgets/text/core/lines/Lines.cpp:49]:
```suggestion
        const Length initHeight = text->getTextFormat().getFont()->info.line_height;
```

File: module-gui/gui/widgets/text/core/lines/SingleTextLine.cpp [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        return localCursor.getBlockNumber() != 0 || localCursor.getPosition() != 0;
+    }
+
+    bool SingleTextLine::checkRightEndCondition(BlockCursor &localCursor, unsigned int signsCountToShow)
+    {
+        BlockCursor moveCursor(localCursor.getDocument(),
+                               localCursor.getPosition(),
+                               localCursor.getBlockNumber(),
+                               localCursor->getFormat()->getFont());
+        moveCursor += (signsCountToShow);
```
Comment (mjan-dev) [inline module-gui/gui/widgets/text/core/lines/SingleTextLine.cpp:74]:
```suggestion
        moveCursor += signsCountToShow;
```

File: module-gui/gui/widgets/text/core/lines/SingleTextLine.cpp [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+
+        return !moveCursor.atEnd();
+    }
+
+    // This method need correction - it does not handle block properly but there is no use case in system to
+    // handle it for now.
```
Comment (mjan-dev) [inline module-gui/gui/widgets/text/core/lines/SingleTextLine.cpp:80]: In what way the block is not handled properly? 
Could you please elaborate?

**Review by mjan-dev — APPROVED** [review]
Nicely done. Added minor comments.

File: module-gui/gui/widgets/text/core/cursors/TextInLineCursor.cpp (hunk 1 above)
Comment (PrzeBrudny) [reply module-gui/gui/widgets/text/core/cursors/TextInLineCursor.cpp:38]: sure

File: module-gui/test/test-catch-text/test-gui-TextInLineCursor.cpp (hunk 2 above)
Comment (PrzeBrudny) [reply module-gui/test/test-catch-text/test-gui-TextInLineCursor.cpp:147]: sure

File: module-gui/gui/widgets/text/core/lines/Lines.cpp (hunk 3 above)
Comment (PrzeBrudny) [reply module-gui/gui/widgets/text/core/lines/Lines.cpp:49]: sure

File: module-gui/gui/widgets/text/core/lines/SingleTextLine.cpp (hunk 4 above)
Comment (PrzeBrudny) [reply module-gui/gui/widgets/text/core/lines/SingleTextLine.cpp:74]: sure

File: module-gui/gui/widgets/text/core/lines/SingleTextLine.cpp (hunk 5 above)
Comment (PrzeBrudny) [reply module-gui/gui/widgets/text/core/lines/SingleTextLine.cpp:80]: There is an issue that you need to know in what style you should print ellipsis -> it probably would have to use style from corresponding block and all is written in a way that it should but there is a lot of corner cases that does not need to be resolved right now (or even ever -> I do not see clear use case for it).

**Pull request merged**